tests/test_to_container.py

```diff
@@ -487,83 +485,60 @@ class TestThrowOnMissing:
-        [
-            # to_container: throw_on_missing
-            param(DictConfig("???"), id="dict:missing"),
-            param(DictConfig({"a": "???"}), id="dict:missing_value"),
-            param(DictConfig({"a": {"b": "???"}}), id="dict:nested"),
-            param(ListConfig("???"), id="list:missing"),
-            param(ListConfig(["???"]), id="list:missing_elt"),
-            param(ListConfig(["abc", ["???"]]), id="list:nested"),
-            param(OmegaConf.structured(B), id="structured:missing_field"),
-            param(
-                OmegaConf.structured(Users({"Bond": "???"})),  # type: ignore
-                id="structured:missing_in_dict_field",
-            ),
-        ],
-    )
     def test_throw_on_missing_raises(self, op: Any, cfg: Any) -> None:
         with raises(MissingMandatoryValue):
             op(cfg)
 
+    def test_no_throw_on_missing(self, cfg: Any) -> None:
+        assert OmegaConf.to_container(cfg, throw_on_missing=False) == cfg
+
+
+@mark.parametrize(
+    "src",
+    [
+        param({"foo": "${bar}"}, id="interp_key_error"),
+        param(
+            OmegaConf.create({"missing": "???", "subcfg": {"x": "${missing}"}}).subcfg,
+            id="interp_to_missing_in_dict",
+        ),
+        param(
+            OmegaConf.create({"missing": "???", "subcfg": ["${missing}"]}).subcfg,
+            id="interp_to_missing_in_list",
+        ),
+        param(
+            OmegaConf.structured(NestedInterpolationToMissing).subcfg,
+            id="interp_to_missing_in_structured",
+        ),
+    ],
+)
+class TestResolveBadInterpolation:
     @mark.parametrize(
         "op",
         [
             param(
                 lambda cfg: OmegaConf.to_container(
                     cfg, resolve=True, throw_on_missing=True
                 ),
-                id="to_container",
+                id="throw_on_missing",
+            ),
+            param(
+                lambda cfg: OmegaConf.to_container(
+                    cfg, resolve=True, throw_on_missing=False
+                ),
+                id="no_throw_on_missing",
             ),
             param(OmegaConf.to_object, id="to_object"),
         ],
     )
-    @mark.parametrize(
-        "src",
-        [
-            param({"missing": "???", "subcfg": {"x": "${missing}"}}, id="dict_in_dict"),
-            param({"missing": "???", "subcfg": ["${missing}"]}, id="list_in_dict"),
-            param(NestedInterpolationToMissing, id="interp_in_structured"),
-        ],
-    )
-    def test_interpolation_to_missing_throws(self, op: Any, src: Any) -> None:
-        """
-        Interpolation to missing: Test that InterpolationToMissingValueError
-        is raised when resolve==True and throw_on_missing==True.
-        """
+    def test_resolve_raises(self, op: Any, src: Any) -> None:
         cfg = OmegaConf.create(src)
-        with raises(InterpolationToMissingValueError):
-            op(cfg.subcfg)
+        with raises(InterpolationResolutionError):
```

Comment:
Can you add the specific expected exception type as a test parameter, so that interpolations to missing test explicitly for `InterpolationToMissingValueError`?

Reply:
Done in https://github.com/omry/omegaconf/pull/730/commits/6f696f3b56d2cc5783eb2a490a103fbd00b8ec0b.

Reply:
Hmm it seems like you duplicated the list of test parameters, why? (when using a class it's ok to have a function ignore one of the test parameters, if it's not relevant -- at least it seems better to me than duplicating parameters)

Reply:
> ...at least it seems better to me than duplicating parameters

I guess I got carried away trying to make sure that each function only accepts params that it uses. I'll deduplicate :)

Reply:
Deduplicated in https://github.com/omry/omegaconf/pull/730/commits/e11bb050e8f89dd63ff1178bad7b9249e6280c2f.